lib/api/core/storage/clientAdapter.js

```diff
@@ -0,0 +1,161 @@
+/*
```

Comment:
New ClientAdapter class which is in charge to maintain the index cache and check index/collection existence